File: src/main/scala/kim/dencka/easy/other/PowerOfTwo.scala
package kim.dencka.easy.other

object PowerOfTwo extends App {

  def isPowerOfTwo(n: Int): Boolean = {
    if (n<=0) false else (n & (n-1)) == 0
  }

//  def isPowerOfTwo2(n: Int): Boolean = {
//    @annotation.tailrec
//    def hammingWeight(x: Int, weight: Int = 0): Int =
//      if (x != 0) hammingWeight(x & x - 1, weight + 1) else weight
//
//    val ones = hammingWeight(n)
//    if (n < 0) false
//    else ones == 1
//  }

}

File: src/main/scala/kim/dencka/java/easy/other/PowerOfTwo.java
package kim.dencka.java.easy.other;

class PowerOfTwo {
    public static void main(String[] args) {
        System.out.println(isPowerOfTwo(1));
        System.out.println(isPowerOfTwo(4));
        System.out.println(isPowerOfTwo(8));
        System.out.println(isPowerOfTwo(15));
        System.out.println(isPowerOfTwo(16));
    }

    public static boolean isPowerOfTwo(int n) {
        if (n <= 0) return false;
        return (n & (n - 1)) == 0;
    }
}
